src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncMethodBuilderMemberCollection.cs

```diff
@@ -133,16 +133,40 @@ internal static bool TryCreate(SyntheticBoundNodeFactory F, MethodSymbol method,
                     collection: out collection);
             }
 
+            // TODO: in both nongeneric and generic forks below, we should unify both "custom tasklike" and "Task/Task<T>"
+            // down the same custom-tasklike path. (Should mscorlib ever device to put [AsyncBuilder] on Task then that
+            // will happen anyway).
+
             if (method.IsTaskReturningAsync(F.Compilation))
             {
                 var returnType = (NamedTypeSymbol)method.ReturnType;
                 NamedTypeSymbol builderType;
-                MethodSymbol createBuilderMethod;
-                PropertySymbol taskProperty;
-                bool customBuilder = returnType.IsCustomTaskType(out builderType, out createBuilderMethod);
+                MethodSymbol createBuilderMethod = null;
+                PropertySymbol taskProperty = null;
+
+                object builderArgument;
+                bool customBuilder = returnType.IsCustomTaskType(out builderArgument);
                 if (customBuilder)
                 {
-                    taskProperty = GetCustomTaskProperty(F, builderType);
+                    builderType = builderArgument as NamedTypeSymbol;
+                    if (builderType?.IsErrorType() == true)
+                    {
+                    }
```

Comment:
The way I'd done cascading conditions was "(1) if an error has been reported, (2) otherwise if it's an error but one hasn't been reported, (3) otherwise success". How about I add a comment to the first branch to indicate this?